### ISSUE TITLE
SetNativeControl NullReferenceException >3.6.0.293080

### ISSUE THREAD

**Issue by filipeaugustosantos** [xamarin/Xamarin.Forms#6328]
### Steps to Reproduce

1. Create custom LabelRender:
```
using Xamarin.Forms.Platform.iOS;
using ScanDesigner.Models;
using Xamarin.Forms;
using UIKit;
using CoreGraphics;
using System;
using ScanDesigner.iOS;

[assembly: ExportRenderer(typeof(RoundedLabel), typeof(RoundedLabelRenderer))]

namespace ScanDesigner.iOS
{
    public class TagUiLabel : UILabel
    {

        private UIEdgeInsets EdgeInsets = new UIEdgeInsets(10, 5, 10, 5);
        private UIEdgeInsets InverseEdgeInsets = new UIEdgeInsets(-10, -5, -10, -5);

        public override CoreGraphics.CGRect TextRectForBounds(CoreGraphics.CGRect bounds, nint numberOfLines)
        {
            var textRect = base.TextRectForBounds(EdgeInsets.InsetRect(bounds), numberOfLines);
            return InverseEdgeInsets.InsetRect(textRect);
        }       
        public override void DrawText(CoreGraphics.CGRect rect)
        {
            base.DrawText(EdgeInsets.InsetRect(rect));
        }
    }
    public class RoundedLabelRenderer : LabelRenderer
    {    
        protected override void OnElementChanged(ElementChangedEventArgs<Label> e)
        {
            if (Control == null)
            {
                SetNativeControl(new TagUiLabel());
            }
            base.OnElementChanged(e);                    
            if (e.NewElement != null)
            {
                this.Layer.CornerRadius = 10;
                this.Layer.BorderColor= ColorExtensions.ToCGColor(ColorExtensions.ToColor(UIColor.FromRGB(3, 169, 244))); 
                this.Layer.BackgroundColor = ColorExtensions.ToCGColor(Color.GhostWhite);
                this.Layer.BorderWidth = 1;
            }
        }    
    }  
}
```
2. Update Xamarin.Forms above 3.6.0.293080
3. Run

### Expected Behavior
![IMG_0018](https://user-images.githubusercontent.com/36250590/58445997-54c38580-80d5-11e9-9e88-adacb3a9dd8d.PNG)

### Actual Behavior
NullReferenceException on `SetNativeControl(new TagUiLabel());`
### Environment

```
Microsoft Visual Studio Community 2019
Versão 16.1.1
VisualStudio.16.Release/16.1.1+28922.388
Microsoft .NET Framework
Versão 4.8.03752

Versão Instalada: Community

Visual C++ 2019   00435-60000-00000-AA911
Microsoft Visual C++ 2019

Assistentes do Microsoft Visual C++   1.0
Assistentes do Microsoft Visual C++

Extensão do IntelliCode   1.0
Informações Detalhadas de Extensão do IntelliCode para Visual Studio

Extensão ResourcePackage   1.0
Informações detalhadas sobre a extensão ResourcePackage do Visual Studio

Extensão ResourcePackage   1.0
Informações detalhadas sobre a extensão ResourcePackage do Visual Studio

Extensibility Message Bus   1.1.77 (master@24013d5)
Provides common messaging-based MEF services for loosely coupled Visual Studio extension components communication and integration.

Ferramentas C#   3.1.0-beta4-19266-03+9d80dea7fe1b14043b9b2ac4d0b59ed26f508742
Componentes de C# usados no IDE. Dependendo do seu tipo de projeto e configurações, uma versão diferente do compilador poderá ser usada.

Ferramentas Comuns do Azure   1.10
Fornece serviços comuns para serem usados por serviços móveis do Azure e ferramentas do Microsoft Azure.

Ferramentas do Serviço de Aplicativo do Azure v3.0.0   16.1.429.50124
Ferramentas do Serviço de Aplicativo do Azure v3.0.0

Ferramentas do Visual Basic   3.1.0-beta4-19266-03+9d80dea7fe1b14043b9b2ac4d0b59ed26f508742
Componentes do Visual Basic usados no IDE. Dependendo do seu tipo de projeto e configurações, uma versão diferente do compilador poderá ser usada.

Ferramentas F# do Visual 10.4 para F# 4.6   16.1.0-beta.19253.3+42526fe359672a05fd562dc16a91a43d0fe047a7
Ferramentas F# do Microsoft Visual 10.4 para F# 4.6

Gerenciador de Pacotes do NuGet   5.1.0
NuGet Package Manager in Visual Studio. For more information about NuGet, visit https://docs.nuget.org/

Microsoft JVM Debugger   1.0
Provides support for connecting the Visual Studio debugger to JDWP compatible Java Virtual Machines

Microsoft MI-Based Debugger   1.0
Provides support for connecting Visual Studio to MI compatible debuggers

Mono Debugging for Visual Studio   16.1.1 (2473f22)
Support for debugging Mono processes with Visual Studio.

Pacote de Ferramentas do Application Insights para Visual Studio   9.1.00429.1
Ferramentas do Application Insights para Visual Studio

Pacote do Host do Adaptador de Depuração do Visual Studio Code   1.0
Camada de interoperabilidade para hospedar os adaptadores de depuração do Visual Studio Code no Visual Studio

Pacote do VC do Microsoft Visual Studio   1.0
Pacote do VC do Microsoft Visual Studio

ProjectServicesPackage Extension   1.0
ProjectServicesPackage Visual Studio Extension Detailed Info

VisualStudio.Mac   1.0
Mac Extension for Visual Studio

Xamarin   16.1.0.542 (d16-1@68b985244)
Extensão do Visual Studio para habilitar o desenvolvimento para o Xamarin.iOS e o Xamarin.Android.

Xamarin Designer   16.1.0.418 (remotes/origin/d16-1@5b958bb10)
Visual Studio extension to enable Xamarin Designer tools in Visual Studio.

Xamarin Templates   16.2.112 (4db4af4)
Templates for building iOS, Android, and Windows apps with Xamarin and Xamarin.Forms.

Xamarin.Android SDK   9.3.0.22 (HEAD/8e7764fdf)
Xamarin.Android Reference Assemblies and MSBuild support.
    Mono: mono/mono/2018-08@3cb36842fc4
    Java.Interop: xamarin/java.interop/d16-1@5ddc3e3
    LibZipSharp: grendello/LibZipSharp/d16-1@44de300
    LibZip: nih-at/libzip/rel-1-5-1@b95cf3f
    ProGuard: xamarin/proguard/master@905836d
    SQLite: xamarin/sqlite/3.27.1@8212a2d
    Xamarin.Android Tools: xamarin/xamarin-android-tools/d16-1@acabd26


Xamarin.iOS and Xamarin.Mac SDK   12.10.0.153 (750a879)
Xamarin.iOS and Xamarin.Mac Reference Assemblies and MSBuild support.
```

### Build Logs

No error or warnings in build

**Comment by spouliot** [comment]
`SetNativeControl` is part of XF (not XI) and the version mentioned also match XF

**Comment by samhouts** [comment]
Related to #6368

**Comment by rmarinho** [comment]
@filipeaugustosantos can you provide a small reproduction.

Thanks

**Comment by rmarinho** [comment]
Ah ok, you are getting this when navigating from a Page?

**Comment by samhouts** [comment]
closed by #6415

**Comment by filipeaugustosantos** [comment]
I'm sorry for not replying, I was having trouble accessing my account.
The bug has been marked closed but still displays the error.
I'm sorry again if I'm publishing erroneously, I'm new to github.

I've posted another error related to this: [#6656](https://github.com/xamarin/Xamarin.Forms/issues/6656)
